packages/puppeteer-core/src/api/ElementHandle.ts

```diff
@@ -139,6 +139,37 @@ export interface Point {
 export abstract class ElementHandle<
   ElementType extends Node = Element,
 > extends JSHandle<ElementType> {
+  /**
+   * A given method will have it's `this` replaced with an isolated version of
+   * `this` when decorated with this decorator.
+   *
+   * All changes of isolated `this` are reflected on the actual `this`.
+   *
+   * @internal
+   */
+  static bindIsolatedHandle<
+    This extends ElementHandle<Node>,
+    Args extends unknown[],
+    Ret,
+  >(
+    target: (this: This, ...args: Args) => Promise<Ret>,
+    _: unknown
+  ): typeof target {
+    return async function (...args) {
+      using adoptedThis = await this.frame.isolatedRealm().adoptHandle(this);
+      const result = await target.call(adoptedThis, ...args);
+      // If the function returns `adoptedThis`, then we return `this`.
+      if ((result as unknown) === adoptedThis) {
+        return this as unknown as Awaited<Ret>;
+      }
+      // If the function returns a handle, transfer it into the main realm.
+      if (result instanceof JSHandle) {
+        return await this.frame.mainRealm().transferHandle(result);
```

Comment:
why do we need to transfer it to the main realm? isn't it possible that the handle is requested in the isolated realm originally.

Reply:
You're right. I think before we can progress with this PR, I'll need to implement realm awareness in ElementHandle. I'll resolve this comment once it's ready.

Reply:
Let's do that first and then rebase this PR on the sandbox aware handles?